transcom/mymove
# MB-6112 TIO payment basic service item details

## Description

> As a TIO, I need to be able to see which Basic service items are on the payment request to understand what the contractor is asking to be paid for.
> 
> Given tha I am a TIO looking at the Payment Requests page
>  When I see a payment request card awaiting review and it contains either
>       the Move Management or Counseling service items
>  Then I see a table in that card outlining the amounts and statuses of those
>       service items

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- Uses the component from @jenniferlynparsons PR: https://github.com/transcom/mymove/pull/5608
- **Caveat here is we're currently filtering for the basic service items only until we build support for the shipment related service items**. You may notice that some cards are reviewed but don't have any details since we don't show any shipment related service items.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make server_run
make office_client_run
```

Test
1. Log in as a TIO
2. Click on any payment request
3. Notice the details for basic service items

## Code Review Verification Steps


 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6112) for this change

## Screenshots

**Reviewed with no basic service items (shipment service items not supported yet)**
![image](https://user-images.githubusercontent.com/1522549/104075512-d3e0d100-51c7-11eb-8a02-326e986d4000.png)

**Needs review with service items**
![image](https://user-images.githubusercontent.com/1522549/104075558-f96dda80-51c7-11eb-9bc2-d7e3705c7ecc.png)

**Two payment requests reviewed with service items with collapsed details**
![image](https://user-images.githubusercontent.com/1522549/104075615-2e7a2d00-51c8-11eb-9660-9a619b3993ec.png)

**Two payment requests reviewed with service items with expanded details**
![image](https://user-images.githubusercontent.com/1522549/104075653-49e53800-51c8-11eb-88ed-96759bcfb98d.png)
